### PR TITLE
sink(ticdc): add ProtocolOpen to replace default protocol

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 
Add open-protocol protocol.

part of https://github.com/pingcap/ticdc/issues/3879

### What is changed and how it works?
* Add the open-protocol protocol and use it as an alias for default protocol.
* No more accepting protocols that are not known

### Check List 

Tests 

 - Unit test
 - Integration test (TBD)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @3AceShowHand @asddongmen 

Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

**Review by overvenus — APPROVED** [review]
LGTM

File: pkg/config/mq_sink_protocol.go [hunk 1]
```diff
@@ -31,13 +30,14 @@ const (
...
+	ProtocolOpen
 )
 
 // FromString converts the protocol from string to Protocol enum type.
-func (p *Protocol) FromString(protocol string) {
+func (p *Protocol) FromString(protocol string) error {
 	switch strings.ToLower(protocol) {
 	case "default":
-		*p = ProtocolDefault
+		*p = ProtocolOpen
```
Comment (lance6716) [inline pkg/config/mq_sink_protocol.go:40]: now the input and output by `String()` is different, is this expected? For example, backup the output and apply it to a cluster of lower version

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: pkg/config/mq_sink_protocol.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/config/mq_sink_protocol.go:40]: i will fix it in next pr. I will refine the meta dta from  low version cluster.

**Review by lance6716 — APPROVED** [review]

**Pull request merged**